src/Widgets/TaskFormRevealer.vala

```diff
@@ -0,0 +1,173 @@
+    private Granite.Widgets.DatePicker due_datepicker;
+    private Granite.Widgets.TimePicker due_timepicker;
+
+    public signal void cancel_clicked ();
+    public signal void save_clicked (ECal.Component task);
+    public signal void delete_clicked (ECal.Component task);
+
+    public TaskFormRevealer (ECal.Component task) {
+        Object (task: task);
+    }
+
+    construct {
+        due_switch = new Gtk.Switch ();
+        due_switch.valign = Gtk.Align.CENTER;
+
+        var due_label = new Gtk.Label (_("Schedule:"));
+        due_datepicker = new Granite.Widgets.DatePicker ();
+        due_timepicker = new Granite.Widgets.TimePicker ();
+
+        var due_datetimepicker = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 12);
+        due_datetimepicker.add (due_label);
+        due_datetimepicker.add (due_switch);
+        due_datetimepicker.add (due_datepicker);
+        due_datetimepicker.add (due_timepicker);
+
+        var description_textview = new Gtk.TextView ();
+        description_textview.margin = 12;
+        description_textview.top_margin = description_textview.bottom_margin = 12;
+        description_textview.set_wrap_mode (Gtk.WrapMode.WORD_CHAR);
+        description_textview.accepts_tab = false;
+
+        description_textbuffer = new Gtk.TextBuffer (null);
+        description_textview.set_buffer (description_textbuffer);
+
+        var description_scrolled_window = new Gtk.ScrolledWindow (null, null);
+        description_scrolled_window.hscrollbar_policy = Gtk.PolicyType.EXTERNAL;
+        description_scrolled_window.height_request = 140;
+        description_scrolled_window.add (description_textview);
+
+        var description_frame = new Gtk.Frame (null);
+        description_frame.margin_bottom = 0;
+        description_frame.add (description_scrolled_window);
+
+        var delete_button = new Gtk.Button ();
+        delete_button.sensitive = false;
+        delete_button.label = _("Delete Task");
+        delete_button.halign = Gtk.Align.START;
+        delete_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+
+        var cancel_button = new Gtk.Button ();
+        cancel_button.label = _("Cancel");
+        cancel_button.halign = Gtk.Align.START;
+
+        var save_button = new Gtk.Button ();
+        save_button.label = _("Save Changes");
+        save_button.halign = Gtk.Align.END;
+        save_button.get_style_context ().add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
+
+        var button_box = new Gtk.ButtonBox (Gtk.Orientation.HORIZONTAL);
+        button_box.baseline_position = Gtk.BaselinePosition.CENTER;
+        button_box.set_layout (Gtk.ButtonBoxStyle.END);
+
+        button_box.add (delete_button);
+        button_box.set_child_secondary (delete_button, true);
+        button_box.set_child_non_homogeneous (delete_button, true);
+        button_box.add (cancel_button);
+        button_box.set_child_non_homogeneous (cancel_button, true);
```

Comment:
Usually we do want these buttons to be homogeneous. Is there a reason you set them to not be?

Reply:
Changed the delete button to be homogeneous, while keeping the cancel button as not homogeneous. The reason for this is, I figured the alignment of the cancel button is better, when its not homogeneous:

![Screenshot from 2020-02-01 13-25-44](https://user-images.githubusercontent.com/392542/73592068-755e9d80-44f6-11ea-9c74-3b9daf893c90.png)

If the cancel button is homogeneous too, it feels out of place:

![Screenshot from 2020-02-01 13-25-11](https://user-images.githubusercontent.com/392542/73592076-8c9d8b00-44f6-11ea-9faa-1abda1190ce6.png)